### ISSUE TITLE
Conf files not upgraded on upgrade?

### ISSUE THREAD

**Issue by fflorent** [YunoHost-Apps/mautrix_whatsapp_ynh#31]
Hello!

I am reading the code of the application for packaging another app.

I see in the code of the upgrade that we first remove the final_path directory:
https://github.com/YunoHost-Apps/mautrix_whatsapp_ynh/blob/ede0231ac59aaa1d65a3009bcbdfe333bf5e87fd/scripts/upgrade#L134

A bit later then, we backup the config files:
https://github.com/YunoHost-Apps/mautrix_whatsapp_ynh/blob/ede0231ac59aaa1d65a3009bcbdfe333bf5e87fd/scripts/upgrade#L184-L185

As the `$mautrix_config_path` file is stored in `$final_path`, shouldn't we backup before the final_path directory?

Although I have not tested what I suspect, so I may be wrong

**Comment by Gredin67** [comment]
> I see in the code of the upgrade that we first remove the final_path directory:

this line is just for upgrade from a specific version, don't consider it for packaging a new app.
Which app do you plan to package? I may be able to help.

> As the $mautrix_config_path file is stored in $final_path, shouldn't we backup before the final_path directory?
You may be right. Basically what we want to handle in the longer term are following cases:
- The administrator changed the config file manually from the standard config set at install (with manifest parameters)
- The template config file was changed upstream (more or less parameters)
Currently, if one or the other occurs, then at upgrade checksum differs, the previously used config will be renamed and the user warned. The up-to-date template config file with manifest parameters will be set. And the admin has to adapt it again manually if required.

A system for better handling of config files is being developped for an upcoming YunoHost version. Therefore, I would not spend much time on config file issues before it is out. The most important is that the file follows the upstream template such that install succeeds. Then the user can modify it manually.

The config panel is already in the dev branch!
https://github.com/YunoHost/yunohost-admin/pull/366
https://github.com/YunoHost/yunohost/pull/987

**Comment by fflorent** [comment]
> this line is just for upgrade from a specific version, don't consider it for packaging a new app.

OK!

> Which app do you plan to package? I may be able to help.

I plan to package the mx-puppet-discord app:
 - the application: https://github.com/matrix-discord/mx-puppet-discord
 - the package (WIP): https://github.com/fflorent/matrix-puppet-discord_ynh/tree/testing

The documentation has to be done, but as far as I tested, it already works! (installation, removal, backup, restoration, upgrade). 

There are still warning about… checksum precisely :)

Maybe we can talk in private if you're interested. Code review is highly welcome :).

>  You may be right. Basically what we want to handle in the longer term are following cases:

Actually, I discovered that I was wrong, because that's not the `$final_path` but the `$src_path` that is being removed, so that's perfectly fine!

BTW, `ynh_add_config` performs a checksum computation and stores it ([source](https://github.com/YunoHost/yunohost/blob/b452838a1797641f82778e60f4e50e1722837eba/data/helpers.d/utils#L374)). Good to know :)

> The config panel is already in the dev branch!
> YunoHost/yunohost-admin#366
> YunoHost/yunohost#987

:tada: I hope to play with it relatively soon!

Florent
